src/librustc_mir/interpret/operator.rs

```diff
@@ -195,32 +195,31 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             if let Some(op) = op {
                 let l128 = self.sign_extend(l, left_layout) as i128;
                 let r = self.sign_extend(r, right_layout) as i128;
-                let size = left_layout.size;
+                // We need a special check for overflowing remainder:
+                // "int_min % -1" overflows and returns 0, but after casting things to a larger int
+                // type it does *not* overflow nor give an unrepresentable result!
                 match bin_op {
-                    Rem | Div => {
-                        // int_min / -1
+                    Rem => {
                         if r == -1 && l == (1 << (size.bits() - 1)) {
-                            return Ok((Scalar::from_uint(l, size), true, left_layout.ty));
+                            return Ok((Scalar::from_int(0, size), true, left_layout.ty));
```

Comment:
I think what this used to do for `Rem` is plain wrong -- it returned the wrong result. However, even in release builds, remainder *is* checked for overflow. So, during normal execution in CTFE/Miri, the overflowing case here is unreachable. It *is* reachable in const-prop, which however does not care about the return value, just about whether there is an overflow.

Reply:
With control flow const prop we could stop propping in arms that are only reachable by a failing assert terminator.

Reply:
We could also declare `int_min / -1` and `int_min % -1` to be UB. This is consistent with MIR building *always* checking that, and it should resolve the rem/div part of https://github.com/rust-lang/rust/issues/69020.

Reply:
Though OTOH, it is probably more consistent to handle *all* overflow the same way... so maybe not. I hope to play with this next weekend.